### ISSUE TITLE
training LIDC dataset with maskrcnn

### ISSUE THREAD

**Issue by leowangzi** [MIC-DKFZ/medicaldetectiontoolkit#22]
I trained the LIDC dataset with maskrcnn network, and observed the loss function of training as follows.
![newplot](https://user-images.githubusercontent.com/395896/53609163-8a54f180-3c00-11e9-9741-273c7c8e7376.png)
In the early stage of training, 
dcount[0,0]mrcnn_bbox: 0.00, mrcnn_mask: 0.00, dcount[0,0], 
total loss decreased rapidly.
In the middle stage of training, 
mrcnn_class: 0.75, mrcnn_bbox: 1.05, mrcnn_mask: 0.67, dcount[0, 3]
, causing total Loss to increase suddenly and then decrease slowly.
PS. The default configuration is used on GitHub.
Thanks.

**Comment by wgs123** [comment]
Can you share  the 
LIDC dataset?

**Comment by leowangzi** [comment]
def compute_mrcnn_mask_loss(target_masks, pred_masks, target_class_ids):
    **if 0 not in torch.nonzero(target_class_ids > 0).size():
        # Only positive ROIs contribute to the loss. And only
        # the class specific mask of each ROI.**
        positive_ix = torch.nonzero(target_class_ids > 0)[:, 0]
        positive_class_ids = target_class_ids[positive_ix].long()
        y_true = target_masks[positive_ix, :, :].detach()
        y_pred = pred_masks[positive_ix, positive_class_ids, :, :]
        loss = F.binary_cross_entropy(y_pred, y_true)
    else:
        loss = torch.FloatTensor([0]).cuda()

    return loss

**Comment by pfjaeger** [comment]
hey this is the intuitive behavior of a 2 stage detector. First the RPN needs to train (while the head networks losses are set to 0), once the RPN produces useful detection candidates, these are passed to the head networks for training, so the loss rises.
